### DISCUSSION TITLE
Username autocompletes inside comment blocks

### DISCUSSION THREAD

**Discussion by alexei** [community/community#51918] [Mobile]
The username autocomplete pop up gets triggered inside code blocks which is especially annoying with Python decorators e.g:

```python
@dataclass 
class Foo:
    ...
```

Hitting enter even opens the pop up with all users.

I'm unsure about the benefit of at-mentioning people inside code blocks.

**Comment by iskae** [comment]
Hello @alexei 👋 

Thanks a bunch for reaching out! 

Unfortunately the code blocks are simply markdown editors in which we are attaching our autocomplete logic on both Web and Mobile Apps. In order to disable it within the code blocks we will need to constantly check if the user is currently typing within a code block, which can be a costly process 😅 

Would it be helpful if we provide some sort of switch to disable autocomplete on demand?

**Reply by alexei** [reply]
> In order to disable it within the code blocks we will need to constantly check if the user is currently typing within a code block, which can be a costly process 😅

Fair enough.

> Would it be helpful if we provide some sort of switch to disable autocomplete on demand?

Not for me, 'cause I need it otherwise.

Maybe an option to make it easier to close/discard it? For example, when I type `@asd` the suggestions pop-up opens:

![IMG_0053](https://user-images.githubusercontent.com/96283/229816643-cb9d4611-b2c7-4706-b8de-0796583fda73.jpeg)

Then when I hit `Enter` to move on the next line (unless the decorator matches a username...), it closes the suggestions pop-up but opens another pop-up with all users:

![IMG_0054](https://user-images.githubusercontent.com/96283/229816701-db711139-2358-47db-a4a5-f841e3ced7cc.jpeg)

So while my `Enter`ing the next line was a workaround to close the suggestions box, that created another problem.

If the suggestions pop-up had an alternate mechanism (e.g. an &times; in the top right corner, or just taping outside of it) to close/discard it for the current cursor position, then I guess _that_ would help, because one could use that instead of `Enter`ing.
